libraries/helpers.rb

```diff
@@ -81,14 +81,67 @@ def zap_extra_env_files
       end
     end
 
-    def wait_for_service
-      unless inside_docker?
-        sleep 1 until ::FileTest.pipe?("#{service_dir_name}/supervise/ok")
+    # Checks if runsvdir process is running. runsvdir
+    # starts and monitors services from #{parsed_service_dir}.
+    # (http://smarden.org/runit/runsvdir.8.html). 
+    # True e.g. in a docker container which uses Phusion 
+    # baseimage-docker, but false when running `docker build`. 
+    def runsvdir_running?
+      cmd = "ps -A -o command | grep \"^/usr/bin/runsvdir -P"\
```

Comment:
@coderanger I replaced the `ps ... | grep ...` command with verifying the `/proc/#{pid}/status` file. (See also the commit message).